components/camera/videosource/static.go

```diff
@@ -74,8 +76,12 @@ func (fs *fileSource) Read(ctx context.Context) (image.Image, func(), error) {
 	return img, func() {}, err
 }
 
-// NextPointCloud returns the point cloud from projecting the rgb and depth image using the intrinsic parameters.
+// NextPointCloud returns the point cloud from projecting the rgb and depth image using the intrinsic parameters,
+// or the pointcloud from file if set.
 func (fs *fileSource) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	if fs.PointCloudFN != "" {
+		return pointcloud.NewFromFile(fs.PointCloudFN, nil)
+	}
```

Comment:
I think one thing you can do here is cache this point cloud. So the first time, you can do a NewFromFile, but then you should store the pointcloud in the fileSource struct, and use that on future calls.

so essentially:
```
if fs.PointCloudFN != "" && fs.pc == nil {
   newPc, err := pointcloud.NewFromFile(fs.PointCloudFN, nil),
   if err != nil {
      return nil, err
   }
  fs.pc = newPc
} 
if fs.pc != nil {
  return fs.pc, nil
}
```